tests/FSharp.Compiler.LanguageServer.UnitTests/DiagnosticsTests.fs

```diff
@@ -0,0 +1,152 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace FSharp.Compiler.LanguageServer.UnitTests
+
+open System
+open System.IO
+open System.Linq
+open System.Threading.Tasks
+open FSharp.Compiler.LanguageServer
+open Nerdbank.Streams
+open NUnit.Framework
+
+[<TestFixture>]
+type DiagnosticsTests() =
+
+    let createTestableProject (tfm: string) (sourceFiles: (string * string) list) =
+        let testDir = new TemporaryDirectory()
+        let directoryBuildText = "<Project />"
+        File.WriteAllText(Path.Combine(testDir.Directory, "Directory.Build.props"), directoryBuildText)
+        File.WriteAllText(Path.Combine(testDir.Directory, "Directory.Build.targets"), directoryBuildText)
+        for name, contents in sourceFiles do
+            File.WriteAllText(Path.Combine(testDir.Directory, name), contents)
+        let compileItems =
+            sourceFiles
+            |> List.map fst
+            |> List.map (sprintf "<Compile Include=\"%s\" />")
+            |> List.fold (fun content line -> content + "\n" + line) ""
+        let replacements =
+            [ "{{COMPILE}}", compileItems
+              "{{TARGETFRAMEWORK}}", tfm ]
+        let projectTemplate =
+            @"<Project Sdk=""Microsoft.NET.Sdk"">
+              <PropertyGroup>
+                <OutputType>Library</OutputType>
+                <TargetFramework>{{TARGETFRAMEWORK}}</TargetFramework>
+              </PropertyGroup>
+              <ItemGroup>
+                {{COMPILE}}
+              </ItemGroup>
+            </Project>"
+        let projectFile =
+            replacements
+            |> List.fold (fun (content: string) (find, replace) -> content.Replace(find, replace)) projectTemplate
+        File.WriteAllText(Path.Combine(testDir.Directory, "test.fsproj"), projectFile)
+        testDir
+
+    let createRpcClient (rootPath: string) =
+        let clientStream, serverStream = FullDuplexStream.CreatePair().ToTuple()
+        let server = new Server(serverStream, serverStream)
+        server.StartListening()
+        let client = new TestClient(rootPath, clientStream, clientStream, server)
+        client
+
+    let createClientTest (tfm: string) (sourceFiles: (string * string) list) =
+        let testDir = createTestableProject tfm sourceFiles
+        let client = createRpcClient testDir.Directory
+        client
+
+    let getDiagnostics (content: string) =
+        async {
+            let client = createClientTest "netstandard2.0" [ "Program.fs", content ]
```

Comment:
netcoreapp2.1 perhaps?

Reply:
I made an arbitrary choice for now.  Eventually a test will want to specify a certain TFM, but for these tests that simply ensure diagnostics can flow, I don't care too much.